rusk-prover/src/tx.rs

```diff
@@ -0,0 +1,466 @@
+            })
+            + u64::SIZE
+            + self
+                .call
+                .as_ref()
+                .map(|(_, cname, cdata)| {
+                    CONTRACT_ID_BYTES + u64::SIZE + cname.len() + cdata.len()
+                })
+                .unwrap_or(0);
+
+        let mut buf = vec![0; size];
+        let mut writer = &mut buf[..];
+
+        writer.write(&(num_inputs as u64).to_bytes());
+        for sinput in serialized_inputs {
+            writer.write(&(sinput.len() as u64).to_bytes());
+            writer.write(&sinput);
+        }
+
+        writer.write(&(num_outputs as u64).to_bytes());
+        for soutput in serialized_outputs {
+            writer.write(&soutput);
+        }
+
+        writer.write(&self.anchor.to_bytes());
+        writer.write(&self.fee.to_bytes());
+
+        write_crossover_value_blinder(&mut writer, self.crossover);
+        write_optional_call(&mut writer, &self.call);
+
+        buf
+    }
+
+    /// Deserialize the transaction from a bytes buffer.
+    pub fn from_slice(buf: &[u8]) -> Result<Self, BytesError> {
+        let mut buffer = buf;
+
+        let num_inputs = u64::from_reader(&mut buffer)?;
+        let mut inputs = Vec::with_capacity(num_inputs as usize);
+        for _ in 0..num_inputs {
+            let size = u64::from_reader(&mut buffer)? as usize;
+            inputs.push(UnprovenTransactionInput::from_slice(&buffer[..size])?);
+            buffer = &buffer[size..];
+        }
+
+        let num_outputs = u64::from_reader(&mut buffer)?;
+        let mut outputs = Vec::with_capacity(num_outputs as usize);
+        for _ in 0..num_outputs {
+            let note = Note::from_reader(&mut buffer)?;
+            let value = u64::from_reader(&mut buffer)?;
+            let blinder = JubJubScalar::from_reader(&mut buffer)?;
+
+            outputs.push((note, value, blinder));
+        }
+
+        let anchor = BlsScalar::from_reader(&mut buffer)?;
+        let fee = Fee::from_reader(&mut buffer)?;
+
+        let crossover = read_crossover_value_blinder(&mut buffer)?;
+
+        let call = read_optional_call(&mut buffer)?;
+
+        Ok(Self {
+            inputs,
+            outputs,
+            anchor,
+            fee,
+            crossover,
+            call,
+        })
+    }
+
+    /// Returns the hash of the transaction.
+    pub fn hash(&self) -> BlsScalar {
+        let nullifiers: Vec<BlsScalar> =
+            self.inputs.iter().map(|input| input.nullifier).collect();
+
+        let hash_outputs: Vec<Note> =
+            self.outputs.iter().map(|(note, _, _)| *note).collect();
+        let hash_crossover = self.crossover.map(|c| c.0);
+        let hash_bytes = self.call.clone().map(|c| (c.0.to_bytes(), c.1, c.2));
+
+        Hasher::digest(Transaction::hash_input_bytes_from_components(
+            &nullifiers,
+            &hash_outputs,
+            &self.anchor,
+            &self.fee,
+            &hash_crossover,
+            &hash_bytes,
+        ))
```

Comment:
use `BlsScalar::hash_to_scalar`

Reply:
This would require a more extensive change, and IMO is not in the scope of this PR.